x-pack/plugins/monitoring/server/lib/setup/collection/get_collection_status.js

```diff
@@ -0,0 +1,313 @@
+import { KIBANA_SYSTEM_ID, BEATS_SYSTEM_ID, LOGSTASH_SYSTEM_ID, APM_SYSTEM_ID } from '../../../../../xpack_main/common/constants';
+
+const getRecentMonitoringDocuments = async (req, indexPatterns, clusterUuid) => {
+  const start = get(req.payload, 'timeRange.min', 'now-30s');
+  const end = get(req.payload, 'timeRange.max', 'now');
+
+  const filters = [
+    {
+      range: {
+        'timestamp': {
+          gte: start,
+          lte: end
+        }
+      }
+    }
+  ];
+
+  if (clusterUuid) {
+    filters.push({ term: { 'cluster_uuid': clusterUuid } });
+  }
+
+  const params = {
+    index: Object.values(indexPatterns),
+    size: 0,
+    ignoreUnavailable: true,
+    filterPath: [
+      'aggregations.indices.buckets'
+    ],
+    body: {
+      query: {
+        bool: {
+          filter: filters,
+        }
+      },
+      aggs: {
+        indices: {
+          terms: {
+            field: '_index',
+            size: 50,
+          },
+          aggs: {
+            es_uuids: {
+              terms: {
+                field: 'node_stats.node_id'
+              },
+              aggs: {
+                by_timestamp: {
+                  max: {
+                    field: 'timestamp'
+                  }
+                }
+              }
+            },
+            kibana_uuids: {
+              terms: {
+                field: 'kibana_stats.kibana.uuid'
+              },
+              aggs: {
+                by_timestamp: {
+                  max: {
+                    field: 'timestamp'
+                  }
+                }
+              }
+            },
+            beats_uuids: {
+              terms: {
+                field: 'beats_stats.beat.uuid'
+              },
+              aggs: {
+                by_timestamp: {
+                  max: {
+                    field: 'timestamp'
+                  }
+                }
+              }
+            },
+            logstash_uuids: {
+              terms: {
+                field: 'logstash_stats.logstash.uuid'
+              },
+              aggs: {
+                by_timestamp: {
+                  max: {
+                    field: 'timestamp'
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  };
+
+  const { callWithRequest } = req.server.plugins.elasticsearch.getCluster('monitoring');
+  return await callWithRequest(req, 'search', params);
+};
+
+async function detectProducts(req) {
+  const result = {
+    [KIBANA_SYSTEM_ID]: {
+      doesExist: true,
+    },
+    'elasticsearch': {
+      doesExist: true,
+    },
+    [BEATS_SYSTEM_ID]: {
+      doesExist: false,
+      mightExist: false,
+    },
+    [APM_SYSTEM_ID]: {
+      doesExist: false,
+      mightExist: false,
+    },
+    [LOGSTASH_SYSTEM_ID]: {
+      doesExist: false,
+      mightExist: false,
+    }
+  };
+
+  const msearch = [
+    { index: '*beat-*' },
+    { size: 0, terminate_after: 0 },
```

Comment:
We're going to have to deal with Security issues here. What response will this request return if the user making the request doesn't have access to search against these indices?

Reply:
Good point. We should handle that by just returning false as if we didn't detect any index.

Reply:
I did some testing here and it doesn't seem like any code changes are necessary. The response is same format, just no documents are returned.